engine/runtime/src/main/java/org/enso/interpreter/runtime/builtin/BuiltinFunction.java

```diff
@@ -0,0 +1,31 @@
+package org.enso.interpreter.runtime.builtin;
+
+import org.enso.interpreter.dsl.Owner;
+import org.enso.interpreter.runtime.callable.function.Function;
+
+/** BuiltinFunction encapsulates information about a builtin runtime function and its metadata. */
+public class BuiltinFunction {
```

Comment:
Wouldn't `BuiltinFunction` rather be a `record` than plain `class`?

Reply:
Yes, indeed. Except that frgaal appears to be causing problems when I make that change, as noted in the comment one line below.

Reply:
Yeap, still does it
```
[error] /home/hubert/tmp/enso/engine/runtime/src/main/java/org/enso/interpreter/runtime/builtin/BuiltinFunction.java:7:15: not found: type java
[error] public record BuiltinFunction(Function fun, boolean isStatic, Owner owner) {
[error]               ^
[error] one error found
```
this is a weird one.

Reply:
Is there a simple reproducer that wouldn't involve `sbt`? Then we can report a bug.

Reply:
Seems that might be a known caveat - "record classes" [section](http://frgaal.org/) mentions some problems if the target is lower than 16.